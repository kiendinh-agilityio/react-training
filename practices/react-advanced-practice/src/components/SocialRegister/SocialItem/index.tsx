import { Link } from '@/components/common';

interface SocialItemProps {
  href: string;
  icon: React.ReactNode;
  target?: '_blank' | '_self' | '_parent' | '_top';
}

const SocialItem = ({ href, icon, target = '_blank' }: SocialItemProps) => (
  <li className="list-none flex items-center justify-center bg-light w-[75px] h-[75px] border border-[#e2e8f0] rounded-[15px]">
    <Link href={href} target={target}>
      {icon}
    </Link>
  </li>
);

export default SocialItem;
