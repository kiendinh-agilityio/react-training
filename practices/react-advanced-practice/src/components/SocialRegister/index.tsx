import SocialItem from './SocialItem';

// import common icons
import { FacebookIcon, AppleIcon, GoogleIcon } from '@/components/common/Icons';

export const SocialRegister = () => (
  <ul className="flex gap-4 mt-[22px]">
    <SocialItem href="https://facebook.com" icon={<FacebookIcon />} />
    <SocialItem href="https://www.apple.com/vn/" icon={<AppleIcon />} />
    <SocialItem href="https://www.google.com/" icon={<GoogleIcon />} />
  </ul>
);

export default SocialRegister;
