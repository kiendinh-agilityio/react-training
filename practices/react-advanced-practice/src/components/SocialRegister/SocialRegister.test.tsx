import { render } from '@testing-library/react';
import SocialRegister from '.';

jest.mock('@/components/common/Icons', () => ({
  FacebookIcon: () => <svg data-testid="facebook-icon" />,
  AppleIcon: () => <svg data-testid="apple-icon" />,
  GoogleIcon: () => <svg data-testid="google-icon" />,
}));

describe('SocialRegister', () => {
  test('renders SocialRegister component', () => {
    const { container } = render(<SocialRegister />);

    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});
