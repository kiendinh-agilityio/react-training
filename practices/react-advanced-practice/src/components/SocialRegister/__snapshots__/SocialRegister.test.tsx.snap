// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialRegister renders SocialRegister component 1`] = `
<div>
  <ul
    class="flex gap-4 mt-[22px]"
  >
    <li
      class="list-none flex items-center justify-center bg-light w-[75px] h-[75px] border border-[#e2e8f0] rounded-[15px]"
    >
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto cursor-pointer undefined"
        data-accent-color=""
        href="https://facebook.com"
        target="_blank"
      >
        <svg
          data-testid="facebook-icon"
        />
      </a>
    </li>
    <li
      class="list-none flex items-center justify-center bg-light w-[75px] h-[75px] border border-[#e2e8f0] rounded-[15px]"
    >
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto cursor-pointer undefined"
        data-accent-color=""
        href="https://www.apple.com/vn/"
        target="_blank"
      >
        <svg
          data-testid="apple-icon"
        />
      </a>
    </li>
    <li
      class="list-none flex items-center justify-center bg-light w-[75px] h-[75px] border border-[#e2e8f0] rounded-[15px]"
    >
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto cursor-pointer undefined"
        data-accent-color=""
        href="https://www.google.com/"
        target="_blank"
      >
        <svg
          data-testid="google-icon"
        />
      </a>
    </li>
  </ul>
</div>
`;
